presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveStorageFormats.java

```diff
@@ -332,46 +332,219 @@ public void testSnappyCompressedParquetTableCreatedInHive()
+                                "MICROSECONDS", "1967-01-02 12:34:56.123456",
+                                "NANOSECONDS", "1967-01-02 12:34:56.1234564")),
+                new TimestampAndPrecision(
+                        "NANOSECONDS",
+                        "2020-01-02 12:34:56.1234564",
+                        ImmutableMap.of(
+                                "MILLISECONDS", "2020-01-02 12:34:56.123",
+                                "MICROSECONDS", "2020-01-02 12:34:56.123456",
+                                "NANOSECONDS", "2020-01-02 12:34:56.1234564")),
+                new TimestampAndPrecision(
+                        "NANOSECONDS",
+                        "1967-01-02 12:34:56.1234567",
+                        ImmutableMap.of(
+                                "MILLISECONDS", "1967-01-02 12:34:56.123",
+                                "MICROSECONDS", "1967-01-02 12:34:56.123457",
+                                "NANOSECONDS", "1967-01-02 12:34:56.1234567")),
+                new TimestampAndPrecision(
+                        "NANOSECONDS",
+                        "2020-01-02 12:34:56.1234567",
+                        ImmutableMap.of(
+                                "MILLISECONDS", "2020-01-02 12:34:56.123",
+                                "MICROSECONDS", "2020-01-02 12:34:56.123457",
+                                "NANOSECONDS", "2020-01-02 12:34:56.1234567")),
+                new TimestampAndPrecision(
+                        "NANOSECONDS",
+                        "1967-01-02 12:34:56.123456789",
+                        ImmutableMap.of(
+                                "MILLISECONDS", "1967-01-02 12:34:56.123",
+                                "MICROSECONDS", "1967-01-02 12:34:56.123457",
+                                "NANOSECONDS", "1967-01-02 12:34:56.123456789")),
+                new TimestampAndPrecision(
+                        "NANOSECONDS",
+                        "2020-01-02 12:34:56.123456789",
+                        ImmutableMap.of(
+                                "MILLISECONDS", "2020-01-02 12:34:56.123",
+                                "MICROSECONDS", "2020-01-02 12:34:56.123457",
+                                "NANOSECONDS", "2020-01-02 12:34:56.123456789")));
 
         // insert records one by one so that we have one file per record, which allows us to exercise predicate push-down in Parquet
         // (which only works when the value range has a min = max)
         for (TimestampAndPrecision entry : data) {
-            onHive().executeQuery(format("INSERT INTO %s VALUES (%s, '%s')", tableName, entry.getId(), entry.getValue()));
+            onHive().executeQuery(format("INSERT INTO %s VALUES (%s, '%s')", tableName, entry.getId(), entry.getWriteValue()));
         }
 
+        runTimestampQueries(tableName, data);
+    }
+
+    @Test(dataProvider = "storageFormatsWithNanosecondPrecision", groups = STORAGE_FORMATS)
```

Comment:
storage_formats tests are run multiple times, on many different environments, since file format behavior may interact with things like kerberos, impersonation

these tests do not need to run multiple times, and i guess they take quite some time (correct?)

we can move them to separate group
or, we can have them without a group (then will be run in suite-1, on multine) 

-- in any case it'd be good to document the reason for this in a code comment
or move the tests to a spearate class